### PR TITLE
Add flake8 linter

### PR DESCRIPTION
**Pull request by radujica**
Apart from the first 3 files, all changes are just making me .. er.. the linter happy

Can look at second-to-last 9a63328 commit for the kind of things it caught.

Could also add e.g. pylint but that one is much more picky.

### CODE REVIEW

File: .github/workflows/workflow.yml [hunk 1]
```diff
@@ -35,6 +35,10 @@ jobs:
         run: |
           python -m unittest
 
+      - name: Run linter
+        run: |
+          flake8 --config=setup.cfg u8timeseries
```
Comment (hrzn) [inline .github/workflows/workflow.yml:40]: does this automatically apply the changes on the code?

Comment (endrjuskr) [reply .github/workflows/workflow.yml:40]: @radujica is it failing on some warnings or errors or correct them? I believe former but double-checking :)

Comment (endrjuskr) [reply .github/workflows/workflow.yml:40]: Can we run it before tests or even as separate step so we can run it simultaneously? Unless it is < 2 secs then ignore it

Comment (radujica) [reply .github/workflows/workflow.yml:40]: It's like 1s so easiest to just keep it here.

It doesn't correct anything by itself, just fails the github action if an issue is found (after printing what failed, ofc).

IMO we shouldn't let some external tool edit our code and commit it, e.g. through black lib. It also subtly pushes for a proper dev environment if developing many things ;)

**Comment by radujica** [comment]
> Can we also somehow run it from inside PyCharm?

Guess so: https://gist.github.com/tossmilestone/23139d870841a3d5cba2aea28da1a895

**Pull request merged**